### PR TITLE
[BugFix] compute large or predicate statistics timeout

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
The time-complexity of PredicateStatisticsCalculatingVisitor OR row-count is O(2^n), n is OR number
to compute row count, each child need compute twice:  and-statistics and or-statistics

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 51 / 57 (89.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java|51|57|89.47%|[361, 389, 390, 391, 392, 393]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java [hunk 1]
```diff
@@ -330,4 +349,72 @@ private ColumnStatistic getExpressionStatistic(ScalarOperator operator) {
...
+                Statistics baseStatistics = predicate.getChild(0).accept(this, null);
+                double rowCount = baseStatistics.getOutputRowCount();
+
+                for (int i = 1; i < disjunctive.size(); ++i) {
+                    Statistics orStatistics = disjunctive.get(i).accept(this, null);
+                    rowCount = (baseStatistics.getOutputRowCount() + orStatistics.getOutputRowCount()) / 2;
+                    rowCount = Math.max(rowCount, baseStatistics.getOutputRowCount());
+                    rowCount = Math.max(rowCount, orStatistics.getOutputRowCount());
+                    rowCount = Math.min(rowCount, statistics.getOutputRowCount());
+                    baseStatistics = computeOrPredicateStatistics(baseStatistics, orStatistics, rowCount);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java:384]: Can add the Same short-circuit logic here?

**Review by kangkaisen — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by packy92 — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java [hunk 2]
```diff
@@ -277,11 +295,12 @@ public Statistics visitCompoundPredicate(CompoundPredicateOperator predicate, Vo
...
                     Statistics orItemStatistics = disjunctive.get(i).accept(this, null);
                     Statistics andStatistics = disjunctive.get(i).accept(
-                            new PredicateStatisticsCalculator.PredicateStatisticsCalculatingVisitor(cumulativeStatistics), null);
+                            new PredicateStatisticsCalculatingVisitor(cumulativeStatistics), null);
                     rowCount = cumulativeStatistics.getOutputRowCount() + orItemStatistics.getOutputRowCount() -
                             andStatistics.getOutputRowCount();
                     rowCount = Math.min(rowCount, statistics.getOutputRowCount());
-                    cumulativeStatistics = computeOrPredicateStatistics(cumulativeStatistics, orItemStatistics, rowCount);
+                    cumulativeStatistics =
+                            computeOrPredicateStatistics(cumulativeStatistics, orItemStatistics, rowCount);
```
[the remark this answers is not in the document]
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java:303]: If `rowCount == statistics.getOutputRowCount()`, does it means the Max/Min value of each column in statistics is already right, we don't need change it?

**Review by packy92 — APPROVED** [review]

**Comment by Seaven** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by Seaven** [comment]
https://github.com/Mergifyio backport branch-2.4

**Comment by Seaven** [comment]
https://github.com/Mergifyio backport branch-2.3

**Comment by Seaven** [comment]
https://github.com/Mergifyio backport branch-2.2

**Pull request merged**